invenio_groups/views/settings.py

```diff
@@ -230,23 +253,33 @@ def delete(group_id):
 })
 def members(group_id, page, per_page, q, s):
     """List user group members."""
-    group = Group.query.get(group_id)
-    members = Membership.query_by_group(group_id, with_invitations=True)
-    if q:
-        members = Membership.search(members, q)
-    if s:
-        members = Membership.order(members, Membership.state, s)
-    members = members.paginate(page, per_page=per_page)
-
-    return render_template(
-        "groups/members.html",
-        group=group,
-        members=members,
-        page=page,
-        per_page=per_page,
-        q=q,
-        s=s,
-    )
+    group = Group.query.get_or_404(group_id)
+    if group.can_see_members(current_user) or current_user.is_admin:
+        members = Membership.query_by_group(group_id, with_invitations=True)
+        if q:
+            members = Membership.search(members, q)
+        if s:
+            members = Membership.order(members, Membership.state, s)
+        members = members.paginate(page, per_page=per_page)
+
+        return render_template(
+            "groups/members.html",
+            group=group,
+            members=members,
+            page=page,
+            per_page=per_page,
+            q=q,
+            s=s,
+        )
+    else:
```

Comment:
Personally I would prefer no `else` branch if you've previously finished with `return`.